subprojects/docs/src/docs/userguide/commandLineInterface.adoc

```diff
@@ -0,0 +1,520 @@
+----
+
+Learn more about Java test filtering in the <<test_filtering, Java plugin documentation>>.
+
+This is something you can combine with <<continuous_build, continuous build>> using `--continuous` (or `-t`, for short) to re-execute a subset of tests on every change.
+
+----
+❯ gradle test --continuous --tests "com.mypackage.foo.*"
+----
+
+[[sec:selecting_build]]
+==== Selecting which build to execute
+
+When you run the `gradle` command, it looks for a build file in the current directory. You can use the `-b` option to select another build file. Example:
+
+++++
+<sample id="selectProjectUsingBuildFile" dir="userguide/tutorial/selectProject" title="Selecting the project using a build file">
+    <sourcefile file="subdir/myproject.gradle"/>
+    <output args="-q -b subdir/myproject.gradle hello"/>
+</sample>
+++++
+
+Alternatively, you can use the `-p` option to specify the project directory to use. For multi-project builds you should use `-p` option instead of `-b` option.
+
+++++
+<sample id="selectProjectUsingProjectDir" dir="userguide/tutorial/selectProject" title="Selecting the project using project directory">
+    <output args="-q -p subdir hello"/>
+</sample>
+++++
+
+[[sec:command_line_completion]]
+=== Command-Line Completion
+
+Gradle provides bash and zsh tab completion support for tasks, options, and Gradle properties through https://github.com/gradle/gradle-completion[gradle-completion], installed separately.
+
+++++
+<figure>
+    <title>Gradle Completion</title>
+    <imageobject>
+        <imagedata fileref="img/gradle-completion-4.0.gif"/>
+    </imageobject>
+</figure>
+++++
+
+[[sec:command_line_logging]]
+=== Logging Options
+
+==== Setting Log Level
+You can customize the verbosity of Gradle logging with the following options, ordered from least verbose to most verbose. Learn more in the <<logging, logging documentation>>.
+
+`-Dorg.gradle.logging.level=(quiet warn lifecycle info debug)`::
+Set logging level via Gradle properties.
+
+`-q`, `--quiet`::
+Log errors only.
+
+`-w`, `--warn`::
+Set log level to warn.
+
+`-i`, `--info`::
+Set log level to info.
+
+`-d`, `--debug`::
+Log in debug mode (includes normal stacktrace).
+
+Lifecycle is the default log level.
+
+==== Customizing Log Format
+You can control the use of rich output (colors and font variants) by specifying the "console" mode in the following ways:
+
+`-Dorg.gradle.console=(auto plain rich verbose)`::
+Specify console mode via Gradle properties. Different modes described immediately below.
+
+`--console=(auto plain rich verbose)`::
+Specifies which type of console output to generate.
++
+Set to `plain` to generate plain text only. This option disables all color and other rich output in the console output.
++
+Set to `auto` (the default) to enable color and other rich output in the console output when the build process is attached to a console, or to generate plain text only when not attached to a console.
```

Comment:
:o: Should we maybe explain the default option first in the list?

Reply:
Notes added. Default is different depending on whether Gradle is attached to a terminal, which I'll now explain.

Reply:
done